rust/crd/src/authentication.rs

```diff
@@ -84,70 +116,829 @@ pub fn default_sync_roles_at() -> FlaskRolesSyncMoment {
+            auth_details.resolve_class(client).await
+        };
+        AirflowClientAuthenticationDetailsResolved::resolve(auth_details, resolve_auth_class).await
+    }
+    pub async fn resolve<R>(
+        auth_details: &[AirflowClientAuthenticationDetails],
+        resolve_auth_class: impl Fn(ClientAuthenticationDetails) -> R,
+    ) -> Result<AirflowClientAuthenticationDetailsResolved>
+    where
+        R: Future<Output = Result<AuthenticationClass, stackable_operator::client::Error>>,
+    {
+        let mut resolved_auth_classes: Vec<AirflowAuthenticationClassResolved> = Vec::new();
+        let mut user_registration = None;
+        let mut user_registration_role = None;
+        let mut sync_roles_at = None;
+
+        let mut auth_class_names = BTreeSet::new();
+
+        for entry in auth_details {
+            let auth_class_name = entry.common.authentication_class_name();
+
+            let is_new_auth_class = auth_class_names.insert(auth_class_name);
+            ensure!(
+                is_new_auth_class,
+                DuplicateAuthenticationClassReferencesNotAllowedSnafu { auth_class_name }
+            );
+
+            let auth_class = resolve_auth_class(entry.common.clone())
+                .await
+                .context(AuthenticationClassRetrievalFailedSnafu)?;
+
+            match &auth_class.spec.provider {
+                AuthenticationClassProvider::Ldap(provider) => {
+                    let resolved_auth_class = AirflowAuthenticationClassResolved::Ldap {
+                        provider: provider.to_owned(),
+                    };
+                    if let Some(other) = resolved_auth_classes.first() {
+                        ensure!(
+                            mem::discriminant(other) == mem::discriminant(&resolved_auth_class),
+                            MultipleAuthenticationTypesNotSupportedSnafu
+                        );
+                    }
+
+                    ensure!(
+                        resolved_auth_classes.is_empty(),
+                        MultipleLdapProvidersNotSupportedSnafu
+                    );
+
+                    resolved_auth_classes.push(resolved_auth_class);
+                }
+                AuthenticationClassProvider::Oidc(provider) => {
+                    let resolved_auth_class =
+                        AirflowClientAuthenticationDetailsResolved::from_oidc(
+                            auth_class_name,
+                            provider,
+                            entry,
+                        )?;
+
+                    if let Some(other) = resolved_auth_classes.first() {
+                        ensure!(
+                            mem::discriminant(other) == mem::discriminant(&resolved_auth_class),
+                            MultipleAuthenticationTypesNotSupportedSnafu
+                        );
+                    }
+                    resolved_auth_classes.push(resolved_auth_class);
+                }
+
+                _ => {
```

Comment:
do we need this arm? I think it is better, generally, to specify each possible arm so that if/when we add other options in the future, they don't silently get processed.

Reply:
done with https://github.com/stackabletech/airflow-operator/pull/524/commits/f28c3827abcd6b676268999b0f469d046907e922

Reply:
Thanks!